src/plot_api/plot_api.js

```diff
@@ -2585,6 +2585,9 @@ function makePlotFramework(gd) {
     fullLayout._defs = fullLayout._paper.append('defs')
         .attr('id', 'defs-' + fullLayout._uid);
 
+    fullLayout._topdefs = fullLayout._toppaper.append('defs')
+        .attr('id', 'defs-' + fullLayout._uid);
```

Comment:
@mdtusz I opted for a similar pattern for `<defs>` in the top paper `<svg>` as is currently done in the (bottom) paper `<svg>`

> Then the clipPath should remain positioned relative to the element it's applied to?

After a few quick Google searches, I didn't find anything arguments for or against relative `clipPath` locations.

I think that having one `<defs>` node per `<svg>` does make things a bit clearer. But, above all, I chose this route for consistency with what's currently done in the (bottom) `<svg>`.